include/micm/solver/lu_decomposition.hpp

```diff
@@ -34,12 +33,13 @@ namespace micm
   /// calls to Decompose to do the actual decomposition.
   class LuDecomposition
   {
+    public:
     /// number of elements in the middle (k) loops for lower and upper triangular matrices, respectively,
     /// for each iteration of the outer (i) loop
     std::vector<std::pair<std::size_t, std::size_t>> niLU_;
     /// True when A[i][k] is non-zero for each iteration of the middle (k) loop for the upper
     /// triangular matrix; False otherwise
-    std::vector<bool> do_aik_;
+    std::vector<char> do_aik_;
```

Comment:
why change from `bool` to `char`?

Reply:
I could not use .data() on vector<bool> do_aik to retrieve the memory address of vector because it is a vector<bool> is a special template. The test for functions both on CPU and GPU still passed.

Reply:
The c++ standard chose a poor representation for the `vector<bool>`. The [standard recommends a space-efficient implementation](https://en.cppreference.com/w/cpp/container/vector_bool), so using bits instead of bytes. This is different than how a vector works for other types. I don't really know what this means throughout our code, but I can understand why it may not be supported on the GPU: there would need to be a template specialization for bools only which the NVIDIA engineers probably didn't care to support (rightly)

Reply:
Thanks @K20shores for the clarification. Then probably we just update the comments here to explain why to use `char` instead of `bool`?

Reply:
I could keep it bool for CPU code, and copy the bool vector to char vector separately for GPU. Is that better to eliminate uncertainty?

Reply:
Hi @qinatan , thanks for proposing this idea. As long as the `char` vector works on CPU, I am fine with your current implementation and I would suggest just updating the comments to make it clear.

Reply:
updated comment.